packages/cw1155/src/event.rs

```diff
@@ -0,0 +1,42 @@
+use cosmwasm_std::{attr, Attribute, HumanAddr, Uint128};
+use cw0::Event;
+
+use crate::msg::TokenId;
+
+pub struct TransferEvent<'a> {
+    pub from: Option<&'a HumanAddr>,
+    pub to: Option<&'a HumanAddr>,
+    pub token_id: TokenId,
+    pub amount: Uint128,
+}
+
+impl Event for TransferEvent {
+    fn write_attributes(&self, attributes: &mut Vec<Attribute>) {
```

Comment:
Response object now has an [`add_attribute` method](https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/results/response.rs#L112-L117). Maybe this could make use of it?

```rust
fn write_attributes(&self, resp: &mut Response) {
  resp.add_attributute("action", "transfer");
  resp.add_attributute("token_id", self.token_id);
  // ...
}
```

Not that it does anything different, but seems a bit cleaner to read and sets a nice pattern for other devs. (And I would like your feedback on the method name `add_attribute`, we are still in beta, when it can be changed)

Reply:
I'm not very good at picking english names, I think `add_attribute` is good, I changed the method name in `Event` trait to `add_attribututes` too.
Another possible choice here is `fn iter_attributes(&self, resp: &mut Response) -> impl Iterator<Item=Attribute>`.